src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java

```diff
@@ -363,26 +365,66 @@ public void keyPressed(KeyEvent key){
 	origX = x; 
 	origY = y;
 	switch (key.getKeyChar()){
-	case 'w'	    :	 this.y = this.y - logicHandler.getPlayer().getSpeed(); break;
-	case 'a'	    :  	 this.x = this.x - logicHandler.getPlayer().getSpeed(); break;
-	case 'd'	    :	 this.x = this.x + logicHandler.getPlayer().getSpeed(); break;
-	case 's'		:	 this.y = this.y + logicHandler.getPlayer().getSpeed(); break;
-	default			:	return;
+	case 'w': //UP MOVEMENT
+	    this.y = this.y - logicHandler.getPlayer().getSpeed();
+	    break;
+	case 'q': //UP LEFT DIAGONAL MOVEMENT
+	    this.y = this.y - logicHandler.getPlayer().getSpeed();
+	    this.x = this.x - logicHandler.getPlayer().getSpeed();
+	    break;
+	case 'a': //LEFT MOVEMENT
+	    this.x = this.x - logicHandler.getPlayer().getSpeed();
+	    break;
+	case 'e': //UP RIGHT MOVEMENT
+	    this.y = this.y - logicHandler.getPlayer().getSpeed(); 
+	    this.x = this.x + logicHandler.getPlayer().getSpeed();
+	    break;
+	case 'd': //RIGHT MOVEMENT
+	    this.x = this.x + logicHandler.getPlayer().getSpeed();
+	    break;
+	case 'z': //DOWN LEFT MOVEMENT
+	    this.y = this.y + logicHandler.getPlayer().getSpeed();
+	    this.x = this.x - logicHandler.getPlayer().getSpeed();
+	    break;
+	case 's': //DOWN MOVEMENT
+	    this.y = this.y + logicHandler.getPlayer().getSpeed();
+	    break;
+	case 'c': //DOWN RIGHT MOVEMENT
+	    this.y = this.y + logicHandler.getPlayer().getSpeed();
+	    this.x = this.x + logicHandler.getPlayer().getSpeed();
+	    break;
+	case 'l': //LINGER BUTTON (Player waits in place)
+	    break;        
+	default	:
+	    return;
 	}
 
-	canvas.clear();
-	drawAllWalls();
-	//Writes last received input.
-	canvas.write(key.getKeyChar(),7,23,RoguePanel.white,RoguePanel.black);
-	moveHero();
-	moveMonster();
-
+	//Tell player they cannot run into a wall, else continue with game progress
+	if(logicHandler.getObject(x,y) instanceof Wall){
+	    canvas.write("     Ouch a Wall  ", 61,23,RoguePanel.yellow,RoguePanel.red);
 
-	checkPlayerStatus();
-	checkAllMonsterStatus();
-	trackDiscovery();
+	    //restores player (x,y) player position if a wall is hit
+	    while (!logicHandler.isGround(x, y)) {
```

Comment:
I don't really understand what this loop is doing...  Could it be an `else` for the previous `if` instead?

And if it's checking if a move is valid and moving you back, could you do this before the move happens in the first place?